angr/procedures/libc/rand.py

```diff
@@ -1,6 +1,14 @@
 import angr
+import random
 
 class rand(angr.SimProcedure):
+    RAND_MAX = 32767
     def run(self):
-        rval = self.state.solver.BVS('rand', 31, key=('api', 'rand'))
-        return rval.zero_extend(self.state.arch.bits - 31)
+        # rval = self.state.solver.BVS('rand', 31, key=('api', 'rand'))
+        # rval = rval.zero_extend(self.state.arch.bits - 31)
+
+        if not hasattr(self.project, 'random'):
+            setattr(self.project, 'random', random.Random())
```

Comment:
absolutely do not set new properties on project this way. If you want persistent data storage use `state.globals`.

Furthermore, this is not a good implementation of rand. If you want this implementation for your code you should write it as an out-of-tree simprocedure and then hook it into your project manually.

Reply:
I mis-understand the original implementation. I remove it now, sorry for that.